src/base/static/components/organisms/site-header.js

```diff
@@ -0,0 +1,398 @@
+      color={props.color}
+      variant={props.variant}
+      onClick={props.onClick}
+    >
+      {props.children}
+    </Button>
+  );
+})(props => ({
+  fontFamily: props.theme.text.navBarFontFamily,
+  fontWeight: 600,
+  marginLeft: "4px",
+  marginRight: "4px",
+
+  [mq[0]]: {
+    width: "100%",
+  },
+}));
+
+const SiteTitle = styled(RegularTitle)(props => ({
+  color: props.theme.text.titleColor,
+  fontFamily: props.theme.text.titleFontFamily,
+  marginBottom: 0,
+  marginLeft: "15px",
+  letterSpacing: "1px",
+}));
+
+const NavLink = styled(props => (
+  <Link href={props.href} rel="internal" className={props.className}>
+    {props.children}
+  </Link>
+))(props => ({
+  display: "flex",
+  textDecoration: "none",
+
+  [mq[1]]: {
+    borderLeft:
+      props.position > 0 ? `solid 1px ${props.theme.brand.primary}` : "none",
+  },
+}));
+
+const NavButtonWrapper = styled("span")(props => ({
+  [mq[1]]: {
+    borderLeft:
+      props.position > 0 ? `solid 1px ${props.theme.brand.primary}` : "none",
+  },
+}));
+
+const LanguagePickerMenu = styled("ul")(props => ({
+  backgroundColor: props.theme.bg.default,
+
+  [mq[0]]: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    width: "100%",
+    marginTop: "8px",
+  },
+  [mq[1]]: {
+    display: props.isLanguageMenuVisible ? "block" : "none",
+    position: "fixed",
+    border: "3px solid rgba(0, 0, 0, 0.05)",
+  },
+}));
+
+const LanguagePickerMenuItem = styled("li")(props => ({
+  textTransform: "uppercase",
+  fontSize: "0.75rem",
+  padding: "8px",
+  "&:hover": {
+    color: props.theme.text.secondary,
+    backgroundColor: props.theme.brand.accent,
+    cursor: "pointer",
+  },
+
+  [mq[0]]: {
+    width: "100%",
+  },
+}));
+
+const LanguagePicker = styled("nav")(props => ({
+  textTransform: "uppercase",
+  "&:hover": {
+    cursor: "pointer",
+  },
+
+  [mq[0]]: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    width: "100%",
+    marginTop: "16px",
+    marginRight: 0,
+  },
+  [mq[1]]: {
+    marginRight: "16px",
+  },
+}));
+
+const LanguageLink = styled(Link)({
+  textDecoration: "none",
+
+  [mq[0]]: {
+    display: "flex",
+    width: "100%",
+    textAlign: "center",
+  },
+});
+
+const RightAlignedContainer = styled("div")(props => ({
+  alignItems: "center",
+
+  [mq[0]]: {
+    display: props.isHeaderExpanded ? "flex" : "none",
+    marginLeft: 0,
+    width: "100%",
+  },
+  [mq[1]]: {
+    display: "flex",
+    marginLeft: "auto",
+  },
+}));
+
+const NavBarHamburger = styled("i")({
+  fontSize: "20px",
+  margin: "0 8px 0 8px",
+  "&:hover": {
+    cursor: "pointer",
+  },
+  "&:before": {
+    fontStyle: "normal",
+    fontFamily: "FontAwesome",
+    content: "'\f0c9'",
+  },
+
+  [mq[1]]: {
+    display: "none",
+  },
+});
+
+const LogoTitleWrapper = styled("div")({
+  [mq[0]]: {
+    display: "flex",
+    alignItems: "center",
+    width: "100%",
+  },
+});
+
+const navItemMappings = {
```

Comment:
I feel like we discussed a better pattern at one point for what are essentially `switch` statements in JSX, but for the life of me I couldn't remember what that was. This `navItemMappings` feels incorrect somehow...

Reply:
Maybe there is a better pattern out there, but I think this pattern is fine for now.

I think there is an eslint bug that is causing these components to throw prop type errors. If you rename the `props` variable to something like `linkProps` and `sidebarProps`, then it should stop the error.